chain/chain/src/chain.rs

```diff
@@ -1469,11 +1447,52 @@ impl Chain {
                     }
                     _ => {}
                 }
-                if let Err(e) = self.save_block_height_processed(block_height) {
-                    warn!(target: "chain", "Failed to save processed height {}: {}", block_height, e);
-                }
-                Err(e)
+                return Err(e);
+            }
+        };
+
+        // 2) apply chunks
```

Comment:
I'd mention - that this is the part where we run the contracts etc - but we still don't mutate any storage.